File: umbrella-insurance-react-ui/src/endpoints/soa/user/v1/requests/CreateUserRequest.ts
export class CreateUserRequest {
    emailAddress?: string;
    phoneNumber?: string;
    username?: string;
    firstName?: string;
    middleName?: string;
    surname?: string;
    dateOfBirth?: string;
    ssn?: string;
    password?: string;
    twoFactorMethod?: string;
    consentedToEmails?: boolean;
    consentedToTexts?: boolean;
    consentedToTermsAndConditions?: boolean;

    constructor(obj?: any) {
        if (obj) {
            if (obj?.emailAddress) {
                this.emailAddress = obj.emailAddress;
            }
            if (obj?.phoneNumber) {
                this.phoneNumber = obj.phoneNumber;
            }
            if (obj?.username) {
                this.username = obj.username;
            }
            if (obj?.firstName) {
                this.firstName = obj.firstName;
            }
            if (obj?.middleName) {
                this.middleName = obj.middleName;
            }
            if (obj?.surname) {
                this.surname = obj.surname;
            }
            if (obj?.dateOfBirth) {
                this.dateOfBirth = obj.dateOfBirth;
            }
            if (obj?.ssn) {
                this.ssn = obj.ssn;
            }
            if (obj?.password) {
                this.password = obj.password;
            }
            if (obj?.twoFactorMethod) {
                this.twoFactorMethod = obj.twoFactorMethod;
            }
            if (obj?.consentedToEmails) {
                this.consentedToEmails = obj.consentedToEmails;
            }
            if (obj?.consentedToTexts) {
                this.consentedToTexts = obj.consentedToTexts;
            }
            if (obj?.consentedToTermsAndConditions) {
                this.consentedToTermsAndConditions = obj.consentedToTermsAndConditions;
            }
        }
    }
}

File: umbrella-insurance-backend/src/main/java/com/umbrella/insurance/endpoints/rest/users/v1/requests/CreateUserRequest.java
package com.umbrella.insurance.endpoints.rest.users.v1.requests;

import com.umbrella.insurance.core.deserializers.DateDeserializer;
import com.umbrella.insurance.core.serializers.DateSerializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import lombok.Data;

import java.sql.Date;

@Data
public class CreateUserRequest {
    private String emailAddress;
    private String phoneNumber;
    private String username;
    private String firstName;
    private String middleName;
    private String surname;
    @JsonSerialize(using = DateSerializer.class)
    @JsonDeserialize(using = DateDeserializer.class)
    private Date dateOfBirth;
    private String ssn;
    private String password;
    private String twoFactorMethod;
    private Boolean consentedToEmails;
    private Boolean consentedToTexts;
    private Boolean consentedToTermsAndConditions;
}
